File: Miscellaneous/Golang/26. Remove Duplicates from Sorted Array.go
package main

import (
	"fmt"
)

func main() {

	nums := []int{1, 1, 2}
	result := removeDuplicates(nums)
	fmt.Println(result)
}

func removeDuplicates(arr []int) int {
	keys := make(map[int]bool)
	var ans []int
	for i := 0; i < len(arr); i++ {
		if _, ok := keys[arr[i]]; !ok {
			keys[arr[i]] = true
			ans = append(ans, arr[i])
		}
	}

	return len(ans)
}

File: Miscellaneous/Python/26. Remove Duplicates from Sorted Array.py
class Solution(object):
    def removeDuplicates(self, nums):
        dict = {}
        counter = 0

        for i in range(len(nums)):
           if nums[i] not in dict.keys():
               dict[nums[i]] = True
               counter += 1

    
        return counter

obj = Solution()
arr = [1,1,2]
ans = obj.removeDuplicates(arr)
print(ans)
